### DISCUSSION TITLE
online demo and face/people recognition

### DISCUSSION THREAD

**Discussion by andreap-** [photoprism/photoprism#1527] [General]
like many others I'm very interested in this feature, so today I decided to make a few tests on the official online demo
I checked for PEOPLE, than I choose a person and made a new search for ROLAND.
looks like lots of Roland's portraits do not appear in the results.
Maybe because if I just search PEOPLE I find one of Roland's portraits as the cover of an album/stacked-photos-group _(don't know how to properly name this stacked photos)_ named Andrea?
.
I'm just trying to figure out how accurate the implementation of the people recognition is at the actual stage.

regards

**Comment by lastzero** [comment]
Automatic clustering works much better with many photos in general, most users will have more photos than our demo. Those clusters might also be "damaged" by users testing the feature by setting a person, then saying it's someone else, back and forth... Face detection is a separate issue, some faces are not found yet automatically and we'll improve on that together with the author of the library we use for this. Finding a person by name is possible, even by family name to find members of the same family.

**Comment by lastzero** [comment]
It's true we detect faces in the primary file of stacks only, mainly for performance reasons. Same wise, we don't analyze every frame of a video. This can be changed in the future should this become an issue, maybe with a setting for those who need it.
